pkg/collector/corechecks/servicediscovery/servicediscovery.go

```diff
@@ -116,7 +116,7 @@ func newCheck() check.Check {
 
 // Configure parses the check configuration and initializes the check
 func (c *Check) Configure(senderManager sender.SenderManager, _ uint64, instanceConfig, initConfig integration.Data, source string) error {
-	if !pkgconfig.Datadog().GetBool("service_discovery.enabled") {
+	if !pkgconfig.SystemProbe.GetBool("discovery.enabled") {
```

Comment:
Earlier today this PR was merged https://github.com/DataDog/datadog-agent/pull/27971
you should change it to 
```suggestion
	if !pkgconfig.SystemProbe().GetBool("discovery.enabled") {
```